### PR TITLE
generalize correlation plugins

### LINKED ISSUE #85: Correlations Backend: The compute plugins (particularly correlations) should support more than just microbiome data
Right now our compute plugins assume theyre receiving microbiome abundance data. We're going to have to introduce a means for plugins to discern what data type theyre being handed and make the appropriate objects and call the appropriate methods as needed for that data type. There was a temporary fix added to the two correlation plugins to shift values to always be non-negative for now. that will need to be removed once this ticket is resolved.

its possible well need to rely on variable and collection annotations for this, possibly introduce compute level constraints as part of this ticket, and maybe even new annotations. all of that is still TBD.

### PR DESCRIPTION
**Pull request by d-callan**
depends on https://github.com/VEuPathDB/veupathUtils/pull/39
depends on https://github.com/VEuPathDB/microbiomeComputations/pull/88

resolves https://github.com/VEuPathDB/service-eda-compute/issues/85

### CODE REVIEW

**Comment by d-callan** [comment]
so this still feels a bit hacky to me, but its way better than it was. i dont have the prefilters working yet, but im thinking that can be a separate pr, bc i want to think about how to best do that yet.

also, i notice the values for member types are things like 'Host Eigengene' and i was expecting 'eigengene' regardless of host vs pathogen. i assume this is bc were also using these as display labels around places? if so, that makes me a bit uncomfortable that some fundamental logic here is based on completely free text strings as opposed to something w controlled values like at min an enum.

**Comment by asizemore** [comment]
no the member should be 'eigengene'. I can start that change.

File: src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassayself/CorrelationAssaySelfPlugin.java [hunk 1]
```diff
@@ -102,18 +104,38 @@ protected void execute() {
...
+      // i think we cross that bridge when we get there and know more.. 
+      // NOTE: getMember tells us the member type, rather than gives us a literal member
+      String collectionMemberType = collection.getMember() == null ? "unknown" : collection.getMember();
+      boolean isEigengene = false;
+      if (collectionMemberType.contains("Eigengene")) {
+        isEigengene = true;
+      }
+      
+      if (isEigengene)  {
+        connection.voidEval("assayData <- assayData[order(" + computeEntityIdColName + ")]; " +
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassayself/CorrelationAssaySelfPlugin.java:122]: why order?

File: src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java [hunk 2]
```diff
@@ -177,37 +179,52 @@ protected void execute() {
...
-                                                          proportionNonZeroThresholdRParam +
-                                                          varianceThresholdRParam +
-                                                          stdDevThresholdRParam +
-                                                          ", verbose=TRUE)");
-
-
-      String statsCmd = "writeStatistics(computeResult, NULL, TRUE)";
+      if (isEigengene)  {
+        connection.voidEval("abundanceData <- abundanceData[order(" + computeEntityIdColName + ")]; " + 
+          "abundanceData <- abundanceData[, -as.character(" + dotNotatedIdColumnsString+ "), with=FALSE];" +
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java]: can we change this to 'assayData'? In the assaySelf version this naming distinction made it really clear what should be what.

File: src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassayassay/CorrelationAssayAssayPlugin.java [hunk 3]
```diff
@@ -173,41 +176,56 @@ protected void execute() {
...
+      // presumably as we support more types in the future, this logic will become more complicated?
+      // might even involve subclassing plugins?
+      // i think we cross that bridge when we get there and know more.. 
+      // NOTE: getMember tells us the member type, rather than gives us a literal member
+      String collection1MemberType = collection1.getMember() == null ? "unknown" : collection1.getMember();
+      String collection2MemberType = collection2.getMember() == null ? "unknown" : collection2.getMember();
+      boolean isEigengene = false;
+      if (collection1MemberType.contains("Eigengene") || collection2MemberType.contains("Eigengene")) {
+        if (!(collection1MemberType.contains("Eigengene") && collection2MemberType.contains("Eigengene"))) {
+          throw new IllegalArgumentException("Both collection variables must have the same member type.");
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassayassay/CorrelationAssayAssayPlugin.java]: Is this just for now? Generally this statement is not true.

**Review by asizemore — COMMENTED** [review]
Had some questions but overall looks good. Are there parts that feel hacky to you outside the "member" values?

**Comment by asizemore** [comment]
https://github.com/VEuPathDB/ApiCommonData/commit/bee5e2c350439261db73c9309dd4808e25199143
Once JB patches the table we'll need to clear cache and will be good to go

**Pull request merged**